sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,236 @@
+import { ProcessSummary } from "@sims/utilities/logger";
+import { Repository, UpdateResult } from "typeorm";
+import { CASService } from "@sims/integrations/cas/cas.service";
+import { CustomNamedError } from "@sims/utilities";
+import {
+  CASAuthDetails,
+  CASSupplierResponse,
+  CASSupplierResponseItem,
+  CASSupplierResponseItemAddress,
+} from "@sims/integrations/cas/models/cas-supplier-response.model";
+import { CAS_AUTH_ERROR } from "@sims/integrations/constants";
+
+const CAS_SUPPLIER_ADDRESS_ACTIVE_STATUS = "ACTIVE";
+
+@Injectable()
+export class CASSupplierIntegrationService {
+  constructor(
+    private readonly casService: CASService,
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {}
+
+  /**
+   * CAS integration process.
+   * Logs on CAS supplier API and request the supplier information for the students with pending supplier information.
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.logon();
+      summary.info("Logon successful.");
+      suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+        casSuppliers,
+        summary,
+        auth,
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError && error.name === CAS_AUTH_ERROR) {
+        summary.error(error.message);
+      } else {
+        summary.error("Unexpected error.", error);
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * For each pending CAS supplier, request supplier information to CAS API and update local table.
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    parentProcessSummary: ProcessSummary,
+    auth: CASAuthDetails,
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      const summary = new ProcessSummary();
+      parentProcessSummary.children(summary);
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        supplierResponse = await this.casService.getSupplierInfoFromCAS(
+          auth.access_token,
+          casSupplier.student.sinValidation.sin,
+          casSupplier.student.user.lastName.toUpperCase(),
+        );
+        suppliersUpdated = await this.updateActiveAddress(
```

Comment:
The name `updateActiveAddress` can be misleading since the core goal would be to update the supplier itself. I would recommend something like `updateSupplier`.